src/3p-frame.js

```diff
@@ -294,7 +294,11 @@ function getAdsLocalhost(win) {
   if (adsUrl == 'https://3p.ampproject.net') {
     adsUrl = 'http://ads.localhost'; // local dev with a localhost server
   }
-  return adsUrl + ':' + (win.location.port || win.parent.location.port);
+  return (
+    adsUrl + ':' + new URL(win.document.baseURI)?.port ||
```

Comment:
I thought it should be `adsUrl + ':' + (a || b || c)`.

Tried in devtools
```
':' + 0 || 2
':0'
```

Reply:
good catch, thank you